core/isolate.rs

```diff
@@ -440,17 +440,19 @@ impl<B: Behavior> Future for Isolate<B> {
 
       let mut overflow_response: Option<Buf> = None;
 
+      let start_index = thread_rng().gen::<usize>() % self.pending_ops.len();
```

Comment:
I think you can use https://docs.rs/rand/*/rand/trait.Rng.html#method.gen_range here (may be slightly more efficient)